### PR TITLE
[doc] Update docs

### PR DESCRIPTION
**Pull request by ahartikainen**
#### Summary:

Fix some typos
Fix formatting

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):
Ari Hartikainen
By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: GPLv3 (http://opensource.org/licenses/GPL-3.0)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by riddell-stan** [comment]
This is tied to #692 , right?

**Comment by ahartikainen** [comment]
Not really, these are mostly doc fixes.

I removed JSON comment about inv metric. It is not tested and our "python -> file" algo uses Rdump format.

File: doc/api.rst [hunk 1]
```diff
@@ -10,7 +10,7 @@ API
    stanc
    StanModel
 
-:ref:`StanFit4model <StanFit4model>` instances are also documented on this page.
+:ref:``StanFit4model <StanFit4model>`` instances are also documented on this page.
```
Comment (riddell-stan) [inline doc/api.rst]: This doesn't seem like the right syntax. https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html

Comment (ahartikainen) [reply doc/api.rst]: StanFit4model <StanFit4model>

Is that supposed to be a link to StanFit4model docs?

Comment (ahartikainen) [reply doc/api.rst]: I think the current format is correct

    :ref:`StanFit4model`

**Review by riddell-stan — APPROVED** [review]

**Pull request merged**